osu.Game.Tests/Skins/SkinDeserialisationTest.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.OpenGL.Textures;
+using osu.Framework.Graphics.Textures;
+using osu.Framework.IO.Stores;
+using osu.Game.Audio;
+using osu.Game.IO;
+using osu.Game.IO.Archives;
+using osu.Game.Skinning;
+using osu.Game.Tests.Resources;
+
+namespace osu.Game.Tests.Skins
+{
+    /// <summary>
+    /// Test that the main components (which are serialised based on namespace/class name)
+    /// remain compatible with any changes.
+    /// </summary>
+    /// <remarks>
+    /// If this test breaks, check any naming or class structure changes.
+    /// Migration rules may need to be added to <see cref="Skin"/>.
+    /// </remarks>
+    [TestFixture]
+    public class SkinDeserialisationTest
```

Comment:
May be worthwhile to have this test also fail when new skinnable drawables are added to the game (likely using reflection), to cover against them when they're later on renamed or otherwise.

Reply:
I've added quite a bit more coverage.